pkg/agent/util/ipset/ipset_test.go

```diff
@@ -0,0 +1,197 @@
+// Copyright 2024 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ipset
+
+import (
+	"errors"
+	"slices"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"k8s.io/utils/exec"
+	exectesting "k8s.io/utils/exec/testing"
+)
+
+func generateFakeOutputFn(stdout, stderr []byte, err error) exectesting.FakeAction {
+	return func() ([]byte, []byte, error) {
+		return stdout, stderr, err
+	}
+}
+
+func assertFakeCmdCall(t *testing.T, outputFn exectesting.FakeAction, expectedCommand string, expectedArgs ...string) exectesting.FakeCommandAction {
+	if outputFn == nil {
+		outputFn = func() ([]byte, []byte, error) {
+			return nil, nil, nil
+		}
+	}
+	return func(cmd string, args ...string) exec.Cmd {
+		if expectedCommand != cmd {
+			t.Errorf("Wrong cmd called: got %v, expected: %v", cmd, expectedCommand)
+		}
+		if !slices.Equal(args, expectedArgs) {
+			t.Errorf("Wrong args: got %v, expected %v", args, expectedArgs)
+		}
```

Comment:
I always find the exec testing package a bit confusing myself, but I think the way it is meant to be used, you are supposed to use `CombinedOutputLog` for such assertions. I don't know if this is the best reference, but https://github.com/kubernetes/kubernetes/blob/master/pkg/proxy/ipvs/ipset/ipset_test.go has some examples.

Reply:
The reference example looks better.